storage/database/sharded_database.go

```diff
@@ -155,75 +161,251 @@ func (db *shardedDB) Close() {
 	}
 }
 
+// Not enough size of channel slows down the iterator
+const shardedDBCombineChanSize = 1024 // Size of resultCh
+const shardedDBSubChannelSize = 128   // Size of each sub-channel of resultChs
+
+// shardedDBIterator iterates all items of each shardDB.
+// This is useful when you want to get items in serial in binary-alphabetigcal order.
 type shardedDBIterator struct {
-	// TODO-Klaytn implement this later.
-	iterators []Iterator
-	key       []byte
-	value     []byte
+	shardedDBChanIterator
 
-	//numBatches uint
-	//
-	//taskCh   chan pdbBatchTask
-	//resultCh chan pdbBatchResult
+	resultCh chan common.Entry
+	key      []byte // current key
+	value    []byte // current value
 }
 
 // NewIterator creates a binary-alphabetical iterator over a subset
 // of database content with a particular key prefix, starting at a particular
 // initial key (or after, if it does not exist).
-func (pdb *shardedDB) NewIterator(prefix []byte, start []byte) Iterator {
-	// TODO-Klaytn implement this later.
-	return nil
+func (db *shardedDB) NewIterator(prefix []byte, start []byte) Iterator {
+	it := &shardedDBIterator{
+		shardedDBChanIterator: db.NewChanIterator(context.TODO(), prefix, start, nil),
+		resultCh:              make(chan common.Entry, shardedDBCombineChanSize),
+	}
+
+	go it.runCombineWorker()
+
+	return it
+}
+
+// runCombineWorker fetches any key/value from resultChs and put the data in resultCh
+// in binary-alphabetical order.
+func (it *shardedDBIterator) runCombineWorker() {
+
+	// creates min-priority queue smallest values from each iterators
+	entries := &entryHeap{}
+	heap.Init(entries)
```

Comment:
Is `Init` necessary? It looks doing nothing.

Reply:
`entryHeap` inherits `container/heap`. You can checkout https://golang.org/pkg/container/heap/ for explanation and examples.
```
	h := &IntHeap{2, 1, 5}
	heap.Init(h)
	heap.Push(h, 3)
	fmt.Printf("minimum: %d\n", (*h)[0])
	for h.Len() > 0 {
		fmt.Printf("%d ", heap.Pop(h))
	}
```